eventstore/test_whatsapp_actions.py

```diff
@@ -209,6 +209,28 @@ def test_delivery_failed_error(self):
             handle_event(event)
             h.assert_called_once_with(event)
 
+    @override_settings(RAPIDPRO_OPTOUT_FLOW="test-flow-uuid")
+    def test_delivery_failed_on_fallback_channel_error(self):
+        """
+        If the event is of type Failed, and uses the fallback channel,
+        then it should trigger the message delivery failed action
+        """
+        event = Event.objects.create()
+        event.fallback_channel = True
+        event.recipient_id = "27820001001"
+        event.timestamp = datetime.datetime(2018, 2, 15, 11, 38, 20, tzinfo=UTC)
+
+        DeliveryFailures.objects.create(number_of_failures=5, contact_id="27820001001")
+
+        with patch("eventstore.tasks.rapidpro") as p:
+            handle_whatsapp_delivery_error(event)
+
+        p.create_flow_start.assert_called_once_with(
+            extra={"optout_reason": "sms_failure", "timestamp": 1518694700},
+            flow="test-flow-uuid",
+            urns=["whatsapp:27820001001"],
+        )
```

Comment:
I think we also need a test for if there is no `DeliveryFailures` object in the database. In that case, then the `create_flow_start` should not be called, but a new `DeliveryFailures` object should be created in the database, with a `number_of_failures` equal to `1`

Reply:
We also need to create another test, for if the event is a successful delivery event, then the `number_of_failures` on the `DeliveryFailures` needs to be reset to 0.